### ISSUE TITLE
the coding “for index_data in baidu_index.get_index(): ” of main is wrong? because the style of file of baidu_index is params

### ISSUE THREAD

**Issue by lanshelinghun** [lijinliangNiu/Spider-Baidu-Index#1]

**Comment by niu-lijinliang** [comment]
I forget some details about this project. But I remeber it can be run, do you try it?

**Comment by lanshelinghun** [comment]
When I replace your url=“http://index.baidu.com/api/SearchApi/index?”+ \ or url = 'http://index.baidu.com/Interface/api/ptbk?uniqid=%s' % uniqidwith my new url,this coding will not work. "url=http://index.baidu.com/v2/index.html" is my new url.

**Comment by niu-lijinliang** [comment]
why do you replace it? And what's the exception？

**Comment by lanshelinghun** [comment]
If I directly use your coding, the result of return always shows "Wait a moment" rather than the output just as 'keyword','type','area'...and so on. in addition, my website's url of baidu index show "Request URL: http://index.baidu.com/v2/index.html", based on that I choose to replace it. Is  it right or wrong?

**Comment by lijinliangNiu** [comment]
I found some bugs. I will fix them soon.

**Comment by niu-lijinliang** [comment]
> If I directly use your coding, the result of return always shows "Wait a moment" rather than the output just as 'keyword','type','area'...and so on. in addition, my website's url of baidu index show "Request URL: http://index.baidu.com/v2/index.html", based on that I choose to replace it. Is it right or wrong?

I have read it. This program is to write the data into the excel sheet, so it does not print any output like 'keyword'... but 'wait a moment' to indicate that a row of data has been crawled.

**Comment by lanshelinghun** [comment]
Yes, I have already run these coding, everything is OK. Thank you very much. I also have another question, You know the amount of crawl of baidu index is up to 1000 for one account of baidu every day. In order to crawl big data, I want to know how to build a pool of cookies which can work all the time. If you are convenient，Please tell me more detail about the above. Thank you very much.

**Comment by niu-lijinliang** [comment]
Soory, I  used this program just to analyze a small amount of data.  I don't know much about cookies pool, but I think you can modify where parameter cookie are used. and keep updating cookie given from the cookie list.

**Comment by lanshelinghun** [comment]
Thanks for your help. I learn more things about baidu index. However, When I add more keywords( about 100 keywords), the result of running is wrong. Can you deal with this question? Thanks.

**Comment by niu-lijinliang** [comment]
> Thanks for your help. I learn more things about baidu index. However, When I add more keywords( about 100 keywords), the result of running is wrong. Can you deal with this question? Thanks.

What is the exception, please.

**Comment by lanshelinghun** [comment]
The machine said that “TypeError: string indices must be integers“. Whether your coding is suitable to add more than 10 or 20 and even more keywords?

**Comment by niu-lijinliang** [comment]
> The machine said that “TypeError: string indices must be integers“. Whether your coding is suitable to add more than 10 or 20 and even more keywords?

Sorry to be late. I think the code is not wrong, and the keywords you used may be wrong. you can change print('Crawlling, wait a moment please.') to print(index_data) in main function. Then you can confirm which keyword is the problem and make sure there is no problem with these keywords.

**Comment by lanshelinghun** [comment]
> > The machine said that “TypeError: string indices must be integers“. Whether your coding is suitable to add more than 10 or 20 and even more keywords?
> 
> Sorry to be late. I think the code is not wrong, and the keywords you used may be wrong. you can change print('Crawlling, wait a moment please.') to print(index_data) in main function. Then you can confirm which keyword is the problem and make sure there is no problem with these keywords.

Thanks for your reply. In my config.file inculding about 100 keywords and more than 300 city_code, thus I guess that maybe too many keywords in the function meanwhile just one cookie which lead error. In addition, I want to ask you how to change crawl of per-day data to per month, Which maybe decrease memory of  dataset. Thank you very much.

**Comment by niu-lijinliang** [comment]
Baidu Index dosen't provide monthly data, so we have to obtain it through accumulation. Well, I guess you are not good at coding. The easiest implementation of multiple crawlers is that you can make a few more copies and change the cookie and keywords in them, then run them together. Although it is not good enough, it is likely to be effective.
